### ISSUE TITLE
can not create admin user [JIRA: RCS-360]

### ISSUE THREAD

**Issue by jomin090** [basho/riak_cs#1299]
While creating the admin user i get following error:

`# curl -XPOST http://localhost:8080/riak-cs/user -H 'Content-Type: application/json' -d '{"email":"xxxx@gmail.com", "name":"admin"}'
<html><head><title>500 Internal Server Error</title></head><body><h1>Internal Server Error</h1>The server encountered an error while processing this request</body></html>[root@lasvegas-nv-datacenter ~]#
[root@lasvegas-nv-datacenter ~]#
[root@lasvegas-nv-datacenter ~]# curl -XPOST http://64.235.33.143:8080/riak-cs/user -H 'Content-Type: application/json' -d '{"email":"jomyad@gmail.com", "name":"admin"}' -v
- About to connect() to 64.235.33.143 port 8080 (#0)
-   Trying 64.235.33.143... connected
- Connected to 64.235.33.143 (64.235.33.143) port 8080 (#0)
  > POST /riak-cs/user HTTP/1.1
  > User-Agent: curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.19.1 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2
  > Host: 64.235.33.143:8080
  > Accept: _/_
  > Content-Type: application/json
  > Content-Length: 44
  >
  < HTTP/1.1 500 Internal Server Error
  < Vary: Accept
  < Server: Riak CS
  < Location: http://64.235.33.143:8080/riak-cs/user
  < Date: Sat, 05 Mar 2016 21:31:15 GMT
  < Content-Type: text/html
  < Content-Length: 170
  <
- Connection #0 to host 64.235.33.143 left intact
- Closing connection #0
  <html><head><title>500 Internal Server Error</title></head><body><h1>Internal Server Error</h1>The server encountered an error while processing this request</body></html>`

I am using versions:

Version:

riak        riak-cs     stanchion
2.1.1           2.1.0       2.1.0

any help is appreciated.

**Comment by ksauzz** [comment]
Hi jomin090,

This is a check list to verify your system works well to create a user. Please make sure these points:
- Stanchion is running?
- Any error log in `/var/log/stanchion/console.log` or `/var/log/riak_cs/console.log`?
- Riak CS and Stanchion were configured to set your Riak address to `riak_host` in riak-cs.conf and stanchion.conf? [[0](http://docs.basho.com/riakcs/2.1.1/cookbooks/configuration/Configuring-Riak-CS/#Host-and-Port)]
- Riak was configured to use `prefix_multi backend` in riak.conf, or use `riak_cs_kv_multi_backend` in advanced.config? [[1](https://github.com/basho/riak_cs/blob/2.1/RELEASE-NOTES.md#riak-kv-211-usage-note)], [[2](http://docs.basho.com/riakcs/2.1.1/cookbooks/configuration/Configuring-Riak/#The-Proper-Backends-for-Riak-CS)]

**Comment by jomin090** [comment]
- stanchion is running fine,
  /etc/init.d/stanchion status
  stanchion (pid  17274) is running...
- tail -f /var/log/stanchion/console.log
  2016-03-12 16:49:36.474 [warning] <0.177.0>@stanchion_utils:email_available:677 Error occurred trying to check if the address <<"mailexample@gmail.com">> has been registered. Reason: <<"{error,{indexes_not_supported,riak_kv_bitcask_backend}}">>
  2016-03-12 16:49:36.475 [error] <0.213.0> Webmachine error at path "/users" : "Internal Server Error"
  - tail -f /var/log/riak-cs/console.log
    2016-03-12 16:54:56.179 [error] <0.810.0> Webmachine error at path "/riak-cs/user" : {error,{error,{badmatch,{error,malformed_xml}},[{riak_cs_s3_response,xml_error_code,1,[{file,"src/riak_cs_s3_response.erl"},{line,396}]},{riak_cs_s3_response,error_response,1,[{file,"src/riak_cs_s3_response.erl"},{line,273}]},{riak_cs_wm_user,accept_json,2,[{file,"src/riak_cs_wm_user.erl"},{line,119}]},{webmachine_resource,resource_call,3,[{file,"src/webmachine_resource.erl"},{line,186}]},{webmachine_resource,do,3,[{file,"src/webmachine_resource.erl"},{line,142}]},{webmachine_decision_core,resource_call,...},...]}} in riak_cs_s3_response:xml_error_code/1 line 396
- riak_host is properly configured in riak-cs.conf and stanchion.conf
-  cat /etc/riak/advanced.config
  [
  {riak_kv, [
  %% Other configs
  {add_paths, ["/usr/lib64/riak-cs/lib/riak_cs-2.1.0/ebin"]},
  {storage_backend, riak_cs_kv_multi_backend},
  {multi_backend_prefix_list, [{<<"0b:">>, be_blocks}]},
  {multi_backend_default, be_default},
  {multi_backend, [
      {be_default, riak_kv_eleveldb_backend, [
          {total_leveldb_mem_percent, 30},
          {data_root, "/var/lib/riak/leveldb"}
      ]},
      {be_blocks, riak_kv_bitcask_backend, [
          {data_root, "/var/lib/riak/bitcask"}
      ]}
  ]}
  %% Other configs
  ]}

]
.
- I have also tried prefix_multi backend in riak.conf

**Comment by ksauzz** [comment]
According to the error message as follow, it seems like that your Riak instance didn't load backend configurations correctly.

```
<<"{error,{indexes_not_supported,riak_kv_bitcask_backend}}">>
```

Please try followings:
- Restart riak again.
- If app.config and vm.args are located in `/etc/riak`, please rename them.
- Rename `advanced.config` to `advanced.config.tmp` or something, and append `storage_backend = prefix_multi` and  `cs_version = 20100` into the tail of riak.conf.

**Comment by ksauzz** [comment]
Since there's been no response on this issue in two weeks, I'm going to close it. If you need more advice, please comment on the issue.

**Comment by ichandan16** [comment]
We ran into the same problem. Not able to create admin user. Got 500 Internal Server Error.

 curl -XPOST http://100.116.149.35:8080/riak-cs/user   -H 'Content-Type: application/json'   -d '{"email":"admin@admin.com", "name":"admin"}'
<html><head><title>500 Internal Server Error</title></head><body><h1>Internal Server Error</h1>The server encountered an error while processing this request</body></html>

Error Log:

2016-06-29 12:35:45.717 [error] <0.1187.0> Webmachine error at path "/riak-cs/user" : {{error,{error,function_clause,[{riak_cs_s3_response,api_error,[{failed_connect,[{to_address,{"100.116.149.35",8085}},{inet,[inet],closed}]},{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",{wm_reqstate,#Port<0.5476>,[{'content-encoding',"identity"},{'content-type',"application/xml"},{mediaparams,[]},{resource_module,riak_cs_wm_user}],undefined,undefined,"100.116.149.35",{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",undefined,"/riak-cs/user","/riak-cs/user",...},...},...},...],...},...]}},...}

Console Log:

2016-06-29 12:58:24.483 [error] <0.863.0> Webmachine error at path "/riak-cs/user" : {{error,{error,function_clause,[{riak_cs_s3_response,api_error,[{failed_connect,[{to_address,{"100.116.149.35",8085}},{inet,[inet],closed}]},{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",{wm_reqstate,#Port<0.4306>,[{'content-encoding',"identity"},{'content-type',"application/xml"},{mediaparams,[]},{resource_module,riak_cs_wm_user}],undefined,undefined,"100.116.149.35",{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",undefined,"/riak-cs/user","/riak-cs/user",...},...},...},...],...},...]}},...}
2016-06-29 12:58:58.687 [error] <0.881.0> Webmachine error at path "/riak-cs/user" : {{error,{error,function_clause,[{riak_cs_s3_response,api_error,[{failed_connect,[{to_address,{"100.116.149.35",8085}},{inet,[inet],closed}]},{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",{wm_reqstate,#Port<0.5157>,[{'content-encoding',"identity"},{'content-type',"application/xml"},{mediaparams,[]},{resource_module,riak_cs_wm_user}],undefined,undefined,"100.116.149.35",{wm_reqdata,'POST',http,{1,1},"100.116.149.35","100.116.149.35",undefined,"/riak-cs/user","/riak-cs/user",...},...},...},...],...},...]}},...}

Kindly help me.

Thanks,
Chandan
